src/rpc/rpcHandler.ts

```diff
@@ -367,11 +367,48 @@ export class RpcHandler implements IRpcEndpoint {
         // Since we don't want our estimations to depend upon baseFee, we set
         // maxFeePerGas to maxPriorityFeePerGas
         userOperation.maxPriorityFeePerGas = userOperation.maxFeePerGas
+        
+
+        // Check if the nonce is valid
+        // If the nonce is less than the current nonce, the user operation has already been executed
+        // If the nonce is greater than the current nonce, we may have missing user operations in the mempool
+        const currentNonceValue = await this.getNonceValue(userOperation, entryPoint)
+        const [,userOperationNonceValue] = getNonceKeyAndValue(userOperation.nonce)
+
+        let queuedUserOperations: UserOperation[] = [];
+        if (userOperationNonceValue < currentNonceValue) {
+            throw new RpcError(
+                "UserOperation reverted during simulation with reason: AA25 invalid account nonce",
+                ValidationErrors.InvalidFields
+            )
+        } else if (userOperationNonceValue > currentNonceValue) {
+            // Nonce queues are supported only for v7 user operations
```

Comment:
wait, we should keep supporting queued nonces for v0.6 no? users rely on it. it's just that we can't pre-simulate them before adding them to the queue

Reply:
Yep, but v6 bulk simulation is tough. there's no easy way to simulate a sequence of user ops (default simulateHandleOp doesn't support it (https://pimlico-workspace.slack.com/archives/C05TL8DN9T4/p1715259742979159?thread_ts=1715257201.121759&cid=C05TL8DN9T4). So we agreed to keep it for v7 only atm, since it works for lattice.

Reply:
right right, i mean keeping it while not simlulating for v0.6 ops